Doc/whatsnew/3.10.rst

```diff
@@ -733,8 +733,8 @@ especially when forward references or invalid types were involved. Compare::
    StrCache = 'Cache[str]'  # a type alias
    LOG_PREFIX = 'LOG[DEBUG]'  # a module constant
 
-Now the :mod:`typing` module has a special annotation :data:`TypeAlias` to
-declare type aliases more explicitly::
+Now the :mod:`typing` module has a special value :data:`TypeAlias`
```

Comment:
Really minor nit:
```suggestion
Now the :mod:`typing` module has a special type :data:`TypeAlias`
```

Reply:
TypeAlias is a function, not a type.  Please see `Lib/typing.py` line 533.

Reply:
I think in ``typing.py`` , the term ``type`` isn't strictly referring to what Python normally deems a ``type`` (it's kind of confusing in some places) but rather what the type checker deems as a type.

Ref: ``Lib/typing.py`` line 449. ``Union`` is called a "type" in the docs, but it's created the same way ``TypeAlias`` is (via a function decorated with the ``_SpecialForm`` class decorator). Another similar example is line 379 ``Any``, which also uses a decorated function, but is called a "type".

Even then I think you're right here. ``TypeAlias`` isn't really a type, it's considered a special marker/construct. So I'm just plain bike shedding here. Sorry :P, I just wanted to share something I found pretty interesting.